### PR TITLE
Remove unknown (deleted) feature toggles at boot

### PR DESCRIPTION
**Pull request by mkllnk**
#### What? Why?

This came up in:
- https://github.com/openfoodfoundation/openfoodnetwork/pull/10779#discussion_r1183194816

##### Introducing auto-cleaning to feature toggles

When we removed feature toggles in the past, we needed a database migration to remove it from the admin interface. But since all current features are listed in the code, we can assume that all features not listed are old or temporary and can be removed safely.

#### What should we test?

**Before deploy:**

- Visit /admin/feature-toggle/features
- Delete some features.
- Add some fictional features.
- **Deploy this pull request.**
- Visit /admin/feature-toggle/features
- The list should reflect all the current features without additions or deletions.

:bell: **Notice for testers:** From now on, if a branch contains a new feature toggle, it will be added when you stage that branch but it will be removed when you stage another branch. So any special configuration of that toggle, for examples for which users it's activated for, will be lost when deploying another branch. We briefly discussed that in a Delivery Circle meeting and the consensus was that this is actually desired behaviour.

#### Release notes

Changelog Category: Technical changes

The title of the pull request will be included in the release notes.

#### Dependencies

The following PR assumes that we have auto-deletion already:

* #10779

### CODE REVIEW

File: spec/lib/open_food_network/feature_toggle_spec.rb [hunk 1]
```diff
@@ -2,17 +2,17 @@
...
+describe OpenFoodNetwork::FeatureToggle do
+  subject(:feature_toggle) { OpenFoodNetwork::FeatureToggle }
 
-      it "uses Flipper configuration" do
-        Flipper.enable(:foo)
-        expect(FeatureToggle.enabled?(:foo)).to be true
-      end
+  context 'when users are not specified' do
+    it "returns false when feature is undefined" do
+      expect(feature_toggle.enabled?(:foo)).to be false
```
Comment (dacook) [inline spec/lib/open_food_network/feature_toggle_spec.rb:10]: I think `described_class` would have also worked :)

File: spec/lib/open_food_network/feature_toggle_spec.rb [hunk 2]
```diff
@@ -15,4 +15,21 @@
...
   end
+
+  describe ".setup!" do
+    it "created all current features at boot time" do
+      expect(Flipper.features.count)
+        .to eq feature_toggle::CURRENT_FEATURES.count
+    end
+
+    it "adds any missing features" do
+      pending "We don't have features to test with at the moment." if Flipper.features.empty?
```
Comment (dacook) [inline spec/lib/open_food_network/feature_toggle_spec.rb:26]: Nice forward thinking!

File: spec/lib/open_food_network/feature_toggle_spec.rb [hunk 3]
```diff
@@ -15,4 +15,21 @@
       expect(feature_toggle.enabled?(:foo)).to be true
     end
   end
+
+  describe ".setup!" do
+    it "created all current features at boot time" do
+      expect(Flipper.features.count)
+        .to eq feature_toggle::CURRENT_FEATURES.count
```
Comment (dacook) [inline spec/lib/open_food_network/feature_toggle_spec.rb:22]: Perhaps we can map and sort these to arrays that could be compared?

**Review by dacook — APPROVED** [review]
Beautiful!

File: spec/lib/open_food_network/feature_toggle_spec.rb (hunk 1 above)
Comment (mkllnk) [reply spec/lib/open_food_network/feature_toggle_spec.rb:10]: I know, but it didn't read that well.

File: spec/lib/open_food_network/feature_toggle_spec.rb (hunk 3 above)
Comment (mkllnk) [reply spec/lib/open_food_network/feature_toggle_spec.rb]: Great idea, added.

**Review by rioug — APPROVED** [review]
Nice one :+1:

**Comment by drummer83** [comment]
Hi @mkllnk,
I followed the given test instructions. Thanks for providing those! :+1: 

## Before staging
Before staging the PR (features I deleted in the next step are crossed out already):
image

Before staging after adding three new features:
image

## After staging
Immediately after staging:
image

## Results
- Manually added features are removed again. :heavy_check_mark: 
- Deleted features are added again. :heavy_check_mark: 
- Newly added features (which were deleted before) will be fully deactivated. :heavy_check_mark: 
- Disabled features in the list can be activated and changes are applied. :heavy_check_mark: 
- Enabled features in the list can be disabled and changes are applied. :heavy_check_mark: 

I think this is looking good and works as expected. Merging! :rocket: 
Thanks Maikel! :partying_face:

**Pull request merged**